config/routes.rb

```diff
@@ -203,11 +203,20 @@
         put :update_mentor, path: "update-mentor"
 
         collection do
-          get :add, to: "add_participants/base#start"
-
-          namespace :add_participants, path: "add" do
-            resource :type, only: %i[show update]
-            resource :details, only: %i[show update]
+          get :add, to: "add_participants#start"
+
+          scope(
+            controller: "add_participants",
+            path: "add",
+            constraints: {
+              step: Regexp.union(
+                *Schools::AddParticipantForm::STEPS.map { |step| step.to_s.tr("_", "-") },
+              ),
+            },
+          ) do
+            get ":step", action: :show
+            patch ":step", action: :update
+            post :complete
```

Comment:
So we can `get`, `patch` and `post` for each of the steps?

Reply:
No, `get` and `patch` for each step and special `post` to complete.